bootloader/bootloadertest/bootloadertest.go

```diff
@@ -146,3 +163,106 @@ func (b *MockBootloader) SetRecoverySystemEnv(recoverySystemDir string, blVars m
 	b.RecoverySystemBootVars = blVars
 	return nil
 }
+
+// SetRunKernelImageEnabledKernel sets the current kernel "symlink" as returned
+// by Kernel(); returns' a restore function to set it back to what it was
+// before.
+func (b *MockBootloader) SetRunKernelImageEnabledKernel(kernel snap.PlaceInfo) func() {
+	old := b.runKernelImageEnabledKernel
+	b.runKernelImageEnabledKernel = kernel
+	return func() {
+		b.runKernelImageEnabledKernel = old
+	}
+}
+
+// SetRunKernelImageEnabledTryKernel sets the current try-kernel "symlink" as
+// returned by TryKernel(). If set to nil, TryKernel()'s second return value
+// will be false; returns' a restore function to set it back to what it was
+// before.
+func (b *MockBootloader) SetRunKernelImageEnabledTryKernel(kernel snap.PlaceInfo) func() {
+	old := b.runKernelImageEnabledTryKernel
+	b.runKernelImageEnabledTryKernel = kernel
+	return func() {
+		b.runKernelImageEnabledTryKernel = old
+	}
+}
+
+// SetRunKernelImageErrorFunction allows setting an error to be returned for the
+// specified function; it returns a restore function to set it back to what it
+// was before
+func (b *MockBootloader) SetRunKernelImageErrorFunction(f string, err error) func() {
+	// check the function
+	switch f {
+	case "EnableKernel", "EnableTryKernel", "Kernel", "TryKernel", "DisableTryKernel":
+		old := b.runKernelImageMockedErrs[f]
+		b.runKernelImageMockedErrs[f] = err
+		return func() {
+			b.runKernelImageMockedErrs[f] = old
+		}
+	default:
+		panic(fmt.Sprintf("unknown ExtractedRunKernelImageBootloader method %q to mock error for", f))
+	}
+}
+
+// GetRunKernelImageFunctionSnapCalls returns what snap calls were specified
```

Comment:
which snaps were mentioned during execution ...

Reply:
Tweaked this a little